examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,858 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `ConditionalDimension` tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial explains how to create equations that only get executed under given conditions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 1., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 1., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 1., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 1., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 1., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from devito import Dimension, Function, Grid\n",
+    "import numpy as np\n",
+    "\n",
+    "# We define a 10x10 grid, dimensions are x, y\n",
+    "shape = (10, 10)\n",
+    "grid = Grid(shape = shape)\n",
+    "x, y = grid.dimensions\n",
+    "\n",
+    "# Define function 𝑓. We will initialize f's data with ones on its diagonal.\n",
+    "\n",
+    "f = Function(name='f', grid=grid)\n",
+    "f.data[:, :] = np.eye(10)\n",
```

Comment:
`f.data[:]` is enough (you wouldn't use the extended notation with one `:` per dimension if you had had 10 dimensions right?

Reply:
oh.. I see. Thanks.